saleor/dashboard/product/forms.py

```diff
@@ -363,6 +366,13 @@ def clean_seo_description(self):
         )
         return seo_description
 
+    def clean_is_published(self):
+        is_published = self.cleaned_data["is_published"]
+        category = self.data["category"]
+        if not category and is_published:
+            raise ValidationError("You must add cateogry to publish product")
```

Comment:
```suggestion
            raise ValidationError("You must select a category to be able to publish")
```

Reply:
@NyanKiyoshi While there is a typo in "cateogry" and it should say "a category", let's not comment on minor language changes which are doesn't really change the meaning or fix any real problem. There is no difference here between wording "add" vs "select" or "publish" or "be able to publish".